src/numeric/grids.jl

```diff
@@ -27,101 +27,165 @@ nodes(grid::EmptyGrid) = nothing
 n_nodes(grid::EmptyGrid) = 0
 node(grid::EmptyGrid, i::Int) = nothing # fail if i!=1 ?
 
+##########################
+# Grid made of one point #
+##########################
 
-immutable PointGrid{N} <: Grid{N}
-    point::Vector{Float64}
+immutable PointGrid{d} <: Grid{d}
+    point::SVector{d,Float64}
+end
 
-    function (::Type{PointGrid{N}}){N}(point::Vector{Float64})
-        @assert N == length(point)
-        new{N}(point)
-    end
+function (::Type{<:PointGrid})(point::Vector{Float64})
+    d = length(point)
+    PointGrid{d}(SVector{d,Float64}(point...))
 end
-nodes(grid::PointGrid) = grid.point'
+
+nodes(grid::PointGrid) = [grid.point]
 n_nodes(grid::PointGrid) = 1
-node(grid::PointGrid, i::Int) = point # fail if i!=1 ?
+node(grid::PointGrid, i::Int) = grid.point # fail if i!=1 ?
 
 
-immutable UnstructuredGrid{N} <: Grid{N}
-    nodes::Matrix{Float64}
+#####################
+# Unstructured Grid #
+#####################
 
-    function (::Type{UnstructuredGrid{N}}){N}(nodes::Matrix{Float64})
-        @assert N == size(nodes, 2)
-        new{N}(nodes)
-    end
+immutable UnstructuredGrid{d} <: Grid{d}
+    nodes::ListOfPoints{d}
 end
-nodes(grid::UnstructuredGrid) = grid.nodes
-n_nodes(grid::UnstructuredGrid) = size(grid.nodes, 1)
-node(grid::UnstructuredGrid, i::Int) = grid.nodes[i, :] # fail if i!=1 ?
 
+# Old-convention
+function (::Type{UnstructuredGrid{d}})(nodes::Matrix{Float64}) where d
+    N = size(nodes,1)
+    @assert d == size(nodes,2)
+    UnstructuredGrid{d}(reinterpret(Point{d}, nodes, (N,)))
```

Comment:
Do we need a transpose here, or are we ok?

Reply:
Indeed, I fixed that.